db/migrate/20230502083009_create_shoutouts.rb

```diff
@@ -0,0 +1,14 @@
+class CreateShoutouts < ActiveRecord::Migration[7.0]
+  def change
+    create_table :shoutouts do |t|
+      t.text :rich_text, null: false
+      t.string :recipients
```

Comment:
Unfortunately I should to have this fileld because for example if user mention about three users and if I will be have  records without this field I get three similar records in the recognition page where user want/can to edit own shoutouts. So he must inlet all three changing instead of one. It is incomfortable

Reply:
@VitaliiShevchenko please schedule a call with Svitlana to go over this.

Reply:
@VitaliiShevchenko if you add proper relations on model level you can get 

`User.last.mentions` and `Shoutout.last.recipients` for any user and any shoutout

I suggest change models relations to

```
class Shoutout < ApplicationRecord
  belongs_to :user
  belongs_to :time_period
  has_many :shoutout_recipients, dependent: :destroy
  has_many :recipients, through: :shoutout_recipients, source: :user

  validates :rich_text, presence: true
  validates_uniqueness_of :digest
  serialize :recipients, JSON
end

```

```
class User < ApplicationRecord
  # Include default devise modules. Others available are:
  # :confirmable, :lockable, :timeoutable, :trackable, :rememberable, :validatable and :omniauthable
  devise :database_authenticatable, :registerable

  has_many :responses, dependent: :destroy
  has_many :shoutouts
  has_many :shoutout_recipients
  has_many :mentions, through: :shoutout_recipients, source: :shoutout

  MAX_NAME_LENGTH = 15

  validates :email, presence: true, uniqueness: true, format: { with: URI::MailTo::EMAIL_REGEXP }
  validates :password, presence: true
  validates :first_name, :last_name, presence: true, length: { maximum: MAX_NAME_LENGTH }
  passwordless_with :email
  scope :opt_in, -> { where(opt_out: false) }
  scope :ordered, -> { order(first_name: :asc) }
end
```